src/clr-angular/forms/combobox/option.ts

```diff
@@ -38,36 +63,48 @@ export class ClrOption<T> implements OnDestroy {
   }
 
   private initializeSubscription(): void {
-    this.subscription = this.optionSelectionService.valueChanged.subscribe((value: T) => {
-      // Check for null and undefined needed because if the user doesnt assign a value to the option,
-      // all options should not be selected as the value would be null or undefined
-      if (value === null || value === undefined) {
-        this.selected = false;
-      } else if (this.value === value) {
-        // TODO: Render option when current selection is set by the user
-        this.selected = true;
-      } else {
-        this.selected = false;
-      }
-    });
+    this.subscriptions.push(
+      this.optionSelectionService.selectionChanged.subscribe((value: ComboboxModel<T>) => {
+        // I think we can just use the latest current in the updatedSelected method instead of using this value; verify
+        this.updateSelected();
+      })
+    );
+  }
+
+  private updateSelected() {
+    const selection = this.optionSelectionService.selectionModel;
+    // Check for null and undefined needed because if the user doesn't assign a value to the option,
+    // all options should not be selected as the value would be null or undefined
+    if (!selection || selection.model === null || selection.model === undefined) {
```

Comment:
```ts
if (!selection || [null, undefined].includes(selection.model)) { ... }
```

Reply:
Some inherited code here. It's a matter of style, I think. I still prefer the previous, as it is easier to read. Especially about `null` and `undefined`.
I'd personally do it like `!selection || selection.model`, but the more explicit check does not bother me too, and I'd prefer not to refactor it unless I see a clear advantage.